latch/ldata/path.py

```diff
@@ -0,0 +1,271 @@
+        """(Re-)populate this LPath's instance's cache.
+
+        Future calls to most getters will return immediately without making a network request.
+
+        Always makes a network request.
+        """
+        data = execute(
+            gql.gql("""
+            query GetNodeData($path: String!) {
+                ldataResolvePathToNode(path: $path) {
+                    path
+                    ldataNode {
+                        finalLinkTarget {
+                            id
+                            name
+                            type
+                            removed
+                            ldataObjectMeta {
+                                contentSize
+                                contentType
+                            }
+                        }
+                    }
+                }
+            }"""),
+            {"path": self.path},
+        )["ldataResolvePathToNode"]
+
+        if data is None or data["ldataNode"] is None:
+            raise FileNotFoundError(f"No such Latch file or directory: {self.path}")
+
+        self._cache.path = self.path
+
+        final_link_target = data["ldataNode"]["finalLinkTarget"]
+        self._cache.node_id = final_link_target["id"]
+        self._cache.name = final_link_target["name"]
+        self._cache.type = LDataNodeType(final_link_target["type"].lower())
+
+        meta = final_link_target["ldataObjectMeta"]
+        if meta is not None:
+            self._cache.size = (
+                -1 if meta["contentSize"] is None else int(meta["contentSize"])
+            )
+            self._cache.content_type = meta["contentType"]
+
+    def node_id(self, *, load_if_missing: bool = True) -> Optional[str]:
+        match = node_id_regex.match(self.path)
+        if match:
+            return match.group("id")
+
+        if self._cache.node_id is None and load_if_missing:
+            self.load_metadata()
+        return self._cache.node_id
+
+    def name(self, *, load_if_missing: bool = True) -> Optional[str]:
+        if self._cache.name is None and load_if_missing:
+            self.load_metadata()
+        return self._cache.name
+
+    def type(self, *, load_if_missing: bool = True) -> Optional[LDataNodeType]:
+        if self._cache.type is None and load_if_missing:
+            self.load_metadata()
+        return self._cache.type
+
+    def size(self, *, load_if_missing: bool = True) -> Optional[int]:
+        if self._cache.size is None and load_if_missing:
+            self.load_metadata()
+        return self._cache.size
+
+    def content_type(self, *, load_if_missing: bool = True) -> Optional[str]:
+        if self._cache.content_type is None and load_if_missing:
+            self.load_metadata()
+        return self._cache.content_type
+
+    def is_dir(self, *, load_if_missing: bool = True) -> bool:
+        return self.type(load_if_missing=load_if_missing) in dir_types
+
+    def iterdir(self) -> Generator["LPath", None, None]:
```

Comment:
```suggestion
    def iterdir(self) -> Iterator[Self]:
```
`Iterator` is simpler when `Generator` is used with `None` for 2nd and 3rd param

`Self` is better for subclassing